src/datatypes/src/schema.rs

```diff
@@ -87,6 +87,11 @@ impl Schema {
             .map(|index| &self.column_schemas[*index])
     }
 
+    pub fn column_type_by_name(&self, name: &str) -> Option<ConcreteDataType> {
+        self.column_schema_by_name(name)
+            .map(|schema| schema.data_type.clone())
+    }
```

Comment:
```suggestion
    pub fn column_type_by_name(&self, name: &str) -> Option<&ConcreteDataType> {
        self.column_schema_by_name(name)
            .map(|schema| &schema.data_type)
    }
```

BTW, is this method necessary? We could use `schema.column_schema_by_name(name).data_type` to access the data type, though it is a bit verbose.